src/util.rs

```diff
@@ -152,6 +154,46 @@ impl From<anyhow::Error> for RpcError {
 /// A serializable result type for use in RPC responses.
 pub type RpcResult<T> = result::Result<T, RpcError>;
 
+#[derive(Debug, Display, Error)]
+pub(crate) enum BaoValidationError {
+    /// Generic io error. We were unable to read the data.
+    IoError(io::Error),
+    // /// The hash of the data does not match the hash of the outboard.
+    // HashMismatch,
```

Comment:
Why are these commented out/not yet used?

Reply:
I want the validation errors to be a bit more informative in the future. E.g. it should say at what offset a hash mismatch occurs. Currently it just passes through the errors from the bao validation.

But I can remove the commented out stuff if you want.